bt_ddos_shield/blockchain_manager.py

```diff
@@ -50,3 +61,56 @@ def get(self, hotkey: Hotkey) -> Optional[bytes]:
         Get data from blockchain for given user identified by hotkey or None if not found.
         """
         pass
+
+
+class BittensorBlockchainManager(AbstractBlockchainManager):
+    """
+    Bittensor BlockchainManager implementation using commitments of knowledge as storage.
+    """
+
+    def __init__(
+        self,
+        address_serializer: AbstractAddressSerializer,
+        subtensor: bittensor.Subtensor,
+        wallet: bittensor_wallet.Wallet,
+        netuid: int,
+    ):
+        super().__init__(address_serializer)
+
+        self.subtensor = subtensor
+        self.wallet = wallet
+        self.netuid = netuid
+
+    def get(self, hotkey: Hotkey) -> Optional[bytes]:
+        """
+        Get data from blockchain for given user identified by hotkey or None if not found.
+        """
+
+        metadata = get_metadata(
```

Comment:
this has retires in it, which is awesome, but i am not sure the retries being ther are part of any public API contract and if they aren't then we should add tests with a mock subtensor that fails a few times before finally returning the thing. If that's not too much hassle.